### ISSUE TITLE
Manage Multiple Facebook bot with one Core Server Instance

### ISSUE THREAD

**Issue by AbdurRub** [RasaHQ/rasa_core#268]
**Rasa Core version**: 0.8.3

**Python version**:  2.7

**Operating system** (windows, osx, ...): Ubuntu

**Issue**:
is it possible to add multiple page access tokens and app secret in fb credentials file to manage multiple bots using one core backend / core server instance?

**Comment by tmbo** [comment]
The current channel implementation for facebook doesn't support this. nevertheless, this isn't a limitation of core, but rather of the input channel implementation.

**Comment by AbdurRub** [comment]
This functionality was supported in early page access version having token dicts. There using pymessenger now core implemented fbmessenger base messenger. any possible solution to this?

**Comment by AbdurRub** [comment]
what if we insert this code snippet if this will be productive. I am asking before trying because I am unable to test right now.

             try:
               output=request.json
               page_id = output['entry'][0]['id']
               if page_id in self.fb_access_token:
                  messenger = Messenger(self.fb_access_token[page_id], on_new_message)
                  messenger.handle(request.get_json(force=True))

**Comment by tmbo** [comment]
Yes I guess that would work.

**Comment by AbdurRub** [comment]
Yes it works.  Now I have to start server again whenever I add new page Id and access token in credentials or there is some solution in which server automatically load credentials file updated.

**Comment by AbdurRub** [comment]
It was giving on send() arguments , so I removed Response argument , why you guys add it?

     self.messenger_client.send(element.to_dict(),
                                   {"sender": {"id": recipient_id}},
                                   'RESPONSE')

**Comment by tmbo** [comment]
Depends on the version of the underlying facebook library you are using - the newest version requires that last argument `'RESPONSE'` to be present whereas earlier versions did not.
